src/Castle.Core.Tests/DynamicProxy.Tests/InvocationProceedInfoTestCase.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2004-2019 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.DynamicProxy.Tests
+{
+	using System;
+
+	using Castle.DynamicProxy.Tests.Interceptors;
+	using Castle.DynamicProxy.Tests.InterClasses;
+	using Castle.DynamicProxy.Tests.Interfaces;
+
+	using NUnit.Framework;
+
+	[TestFixture]
+	public class InvocationProceedInfoTestCase
```

Comment:
Could we get a unit test with 3 interceptors (i.e. 2 interceptors after the `WithCallbackInterceptor`), that's the one I'd be worried about, even thought the code appears to be fine for that since the index is restored.

Reply:
Sure. Can you say more what exactly worries you about a 3-interceptors scenario?

Reply:
I had to think about the logic to make sure that when inside the continuation the index is restored correctly so you could call `GetProceedInfo` or `Proceed` a second time to move into the 3rd interceptor. I'm realising now that you already test this since it is proceeding to the target so that handles the scenario, however having a test setting up another continuation inside the 2nd interceptor would probably be a good test to make sure GetProceedInfo doesn't get broken in the future if we do replace the index.

Reply:
I added a few additional multi-interceptor tests in https://github.com/castleproject/Core/pull/439/commits/62b3698c44766a6e8eccbd50f96bcd8e04146490. Is this the kinds of tests you were thinking of?

Reply:
Yep, looks good.